src/user.js

```diff
@@ -50,46 +50,51 @@ class User {
   }
 
   checkPantryIngredients(recipe) {
+    console.log(recipe.ingredients);
     let missingIngredients = recipe.ingredients.reduce((notPresent, ingredient) => {
-      let ingredientName = ingredient.name;
-      if(!this.pantry.userIngredients[ingredientName] || ingredient.quantity.amount > this.pantry.userIngredients[ingredientName]) {
-        notPresent.push(ingredientName);
+      let ingredientID = ingredient.id;
+      if(!this.pantry.userIngredients[ingredientID] || ingredient.quantity.amount > this.pantry.userIngredients[ingredientID]) {
+        notPresent.push(ingredientID);
       };
       return notPresent;
     }, []);
+    // console.log(missingIngredients);
     if(missingIngredients.length === 0) {
       return 'You have the ingredients!';
     } else {
-      let missing = missingIngredients.join(' and ');
-      return `You still need more ${missing}.`;
+      let missing = missingIngredients.length;
+      return `You still need ${missing} more ingredients.`;
     };
   }
 
   checkHowMuchMore(recipe) {
     let missingIngredients = recipe.ingredients.reduce((notPresent, ingredient) => {
-      if(!this.pantry.userIngredients[ingredient.name] || ingredient.quantity.amount > this.pantry.userIngredients[ingredient.name]) {
+      if(!this.pantry.userIngredients[ingredient.id] || ingredient.quantity.amount > this.pantry.userIngredients[ingredient.id]) {
         notPresent.push(ingredient);
       };
       return notPresent;
     }, []);
-    // console.log(missingIngredients);
     let difference = missingIngredients.map(ingredient => {
       let neededIng;
-      if (this.pantry.userIngredients[ingredient.name]) {
+      let name = recipe.ingredientsData.find(item => {
+        return item.id === ingredient.id;
+      });
+      name = name.name;
+      if (this.pantry.userIngredients[ingredient.id]) {
         neededIng = {
           id: ingredient.id,
           cost: 0,
-          name: ingredient.name,
+          name: name,
           unit: ingredient.quantity.unit,
-          quantityNeeded: parseFloat(ingredient.quantity.amount - this.pantry.userIngredients[ingredient.name].toFixed(2))
+          quantityNeeded: parseFloat(ingredient.quantity.amount - this.pantry.userIngredients[ingredient.id].toFixed(2))
         }
       } else {
           neededIng = {
-            id: NaN,
+            id: ingredient.id,
```

Comment:
Not ideal that we had to refactor this entire section to look at id's as opposed to names, but I think it was a pretty sussinct fix all things considered.

Reply:
Yeah I think this works for what we need. We should come back to this one at the end and see how we can refactor it into smaller methods, though.